src/units.cpp

```diff
@@ -401,19 +398,21 @@ size_t Units::unitCount() const
 
 double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
 {
+    if (!Units::compatible(units1, units2)) {
+        return 0.0;
+    }
+
```

Comment:
I would like to see a default argument added to this method so that I can actively ignore the compatibility check.

Reply:
What do you expect it to return in that situation?  I'd wondered about getting a scaling factor that included the mismatch of units ... is that what you're talking about, or something else?

Reply:
Yes to just getting the scaling factor without being concerned about the mismatch in units.

Reply:
But what "factor"?  a number independent of base units? or the mismatch? 
Option 1:  
`   1e6 = scalingFactor(millisecond, kilogram)`
Option 2:
`   1e6(kg)/(s) = scalingFactor(millisecond, kilogram)`

Reply:
I don't see how you can return option 2.  Option 1 was what I was expecting.

Reply:
We know the base unit mismatch - it would be easy enough to create a `Units` item that filled the gap.

Reply:
If I wanted that I would expect to call a function `unitsMismatch`.